administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+                              (int)$destHeight, $srcWidth, $srcHeight);
+          }
+          // write resized image to file
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_frames,$dest_qual,$dest_imgtype);
         }
 
-        $dst_img = imagecreate($destWidth, $destHeight);
-
-        if (!is_null($offsetx) && !is_null($offsety))
+        if ($metadata)
+        // copy metadata if needed
         {
-          imagecopyresized( $dst_img, $src_img, 0, 0, $offsetx, $offsety,
-                            $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
-        }
-        else
-        {
-          imagecopyresized( $dst_img, $src_img, 0, 0, 0, 0, $destWidth,
-                            (int)$destHeight, $srcWidth, $srcHeight);
+          $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+          if (!$meta_success)
+          {
+            $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+            return false;
+          }      
         }
-        if(!@imagejpeg($dst_img, $dest_file, $dest_qual))
+        if(!$success)
         {
           // Workaround for servers with wwwrun problem
           $dir = dirname($dest_file);
           JoomFile::chmod($dir, '0777', true);
-          imagejpeg($dst_img, $dest_file, $dest_qual);
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_img,$dest_qual,$dest_imgtype);
+          if ($metadata)
+          {
+            $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+            if (!$meta_success)
+            {
+              $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+              return false;
+            }      
+          }
           JoomFile::chmod($dir, '0755', true);
         }
-        imagedestroy($src_img);
-        imagedestroy($dst_img);
+        foreach ($src_frames as $key => $frame)
+        {
+          imagedestroy($src_frames[$key]['image']);
+          imagedestroy($dst_frames[$key]);
+        }
         break;
       case 'gd2':
+        $debugoutput.='GD2...<br/>';
         if(!function_exists('imagecreatefromjpeg'))
+        // check, if GD is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_INSTALLED');
           return false;
         }
         if(!function_exists('imagecreatetruecolor'))
+        // check, if GD2 is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_NO_TRUECOLOR');
           return false;
         }
-        if($imginfo[2] == 'JPG')
+        // create empty image of specified size
+        $dst_frames = array();
+        if ($anim && $imginfo['animation'] && $imginfo['type'] == 'GIF')
         {
-          $src_img = imagecreatefromjpeg($src_file);
+          JLoader::register('GifFrameExtractor', JPATH_COMPONENT_ADMINISTRATOR . '/helpers/GifFrameExtractor.php');
```

Comment:
If the source file is of type GIF and has more than one frame, use GifFrameExtractor Calss to create the src_frames array.